File: algorithms/subtract-the-product-and-sum-of-digits-of-an-integer/subtract-the-product-and-sum-of-digits-of-an-integer.js
/**
 * @param {number} n
 * @return {number}
 */
var subtractProductAndSum = function (n) {
    let chars = n.toString().split('')
    let prod = 1, add = 0
    chars.forEach(c => {
        let num = Number(c)
        prod *= num
        add += num
    })
    return prod - add
};

File: algorithms/subtract-the-product-and-sum-of-digits-of-an-integer/subtract-the-product-and-sum-of-digits-of-an-integer.rs
impl Solution {
    pub fn subtract_product_and_sum(n: i32) -> i32 {
        let s = n.to_string();
        let nums = s.chars().map(|c| c.to_digit(10).unwrap() as i32);
        let add = nums.clone().fold(0, |acc, x| acc + x);
        let prod = nums.clone().fold(1, |acc, x| acc * x);
        
        prod - add
    }
}
